tests/test-extensions/marvel-extension/src/main/java/org/mule/test/marvel/drstrange/DrStrangeOperations.java

```diff
@@ -67,6 +74,93 @@ public String readStream(@Connection MysticConnection connection, @Optional(defa
     }
   }
 
+  public PagingProvider<MysticConnection, Relic> getRelics(StreamingHelper streamingHelper) {
+    return new PagingProvider<MysticConnection, Relic>() {
+
+      private int currentPage = 1;
+      private final int PAGES = 4;
+
+      @Override
+      public List<Relic> getPage(MysticConnection connection) {
+        if (currentPage == PAGES) {
+          return emptyList();
+        }
+        currentPage++;
+        return new ArrayList<Relic>() {
+
+          {
+            add(getResolvedRelic("cloak"));
+            add(getResolvedRelic("boots"));
+            add(getResolvedRelic("staff"));
+          }
+        };
+      }
+
+      private Relic getResolvedRelic(String description) {
+        return new Relic(streamingHelper.resolveCursorProvider(new ByteArrayInputStream(description.getBytes())));
+      }
+
+      @Override
+      public java.util.Optional<Integer> getTotalResults(MysticConnection connection) {
+        return java.util.Optional.empty();
+      }
+
+      @Override
+      public void close(MysticConnection connection) throws MuleException {
+        // Do nothing.
+      }
+    };
+  }
+
+  public PagingProvider<MysticConnection, Relic> getHugeRelic(StreamingHelper streamingHelper) {
+    return new PagingProvider<MysticConnection, Relic>() {
+
+      private boolean pageRetrieved = false;
+
+      @Override
+      public List<Relic> getPage(MysticConnection connection) {
+        if (pageRetrieved) {
+          return emptyList();
+        }
+        pageRetrieved = true;
+        return new ArrayList<Relic>() {
+
+          {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Move the contents of this initializer to a standard constructor or to field initializers. ([squid:S1171](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1171))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7751&issues=AWm78GvHotIXnmsUmYjO&open=AWm78GvHotIXnmsUmYjO)

[ISSUE_KEY]: AWm78GvHotIXnmsUmYjO